### PR TITLE
Support generating a UUID string as the anonymous namespace for Java worker.

### PR DESCRIPTION
**Pull request by jovany-wang**
## Why are these changes needed?
For Java worker, we generate a UUID string as the namespace if a job is not specified a namespace by user. 

## Related issue number
#16474

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: java/runtime/src/main/java/io/ray/runtime/config/RayConfig.java [hunk 1]
```diff
@@ -121,7 +122,8 @@ public RayConfig(Config config) {
     }
 
     // Namespace of this job.
-    namespace = config.getString("ray.job.namespace");
+    String localNamespace = config.getString("ray.job.namespace");
+    namespace = StringUtils.isEmpty(localNamespace) ? UUID.randomUUID().toString() : localNamespace;
```
Comment (kfstorm) [inline java/runtime/src/main/java/io/ray/runtime/config/RayConfig.java:126]: Does the pattern of generated namespace string match Python?

File: java/test/src/main/java/io/ray/test/AnonymousNamespaceTest.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        AnonymousNamespaceTest.class,
+        () ->
+            Assert.assertThrows(
+                NoSuchElementException.class,
+                () -> {
+                  Ray.getGlobalActor("a").get();
+                }));
+  }
+
+  public static void main(String[] args) throws IOException, InterruptedException {
```
Comment (kfstorm) [inline java/test/src/main/java/io/ray/test/AnonymousNamespaceTest.java:31]: Maybe you can create an inner class in `NamespaceTest` with `main` method, so you can put the test case into `NamespaceTest` and avoid adding a new test file.

File: java/runtime/src/main/java/io/ray/runtime/config/RayConfig.java (hunk 1 above)
Comment (jovany-wang) [reply java/runtime/src/main/java/io/ray/runtime/config/RayConfig.java:126]: One example in java: efd2c1ea-f5b5-46b8-b1c3-3020fe2bfd70

Comment (jovany-wang) [reply java/runtime/src/main/java/io/ray/runtime/config/RayConfig.java:126]: It's  UUID4 as well.

**Review by kfstorm — APPROVED** [review]

**Comment by jovany-wang** [comment]
This PR only changes stuff for Java worker. rllib test failures are unrelated. merging it.

**Pull request merged**